### PR TITLE
Add NAPTR support.

### PR DESCRIPTION
**Pull request by karlism**
This is an attempt to create NAPTR record support. Unfortunately it is not fully working, the code compiles, but adding NAPTR DNS record throws following error message:
```
$ cat dnsconfig.js                                                                              
// Providers:
var REG_NONE = NewRegistrar('none', 'NONE');    // No registrar.
var DNS_BIND = NewDnsProvider('bind', 'BIND', {
        'default_soa': {
            'master': 'master.example.com.',
            'mbox': 'hostmaster.example.com.',
            'refresh': 3600,
            'retry': 600,
            'expire': 604800,
            'minttl': 1440,
        },
        'default_ns': [
            'ns1.example.com.',
            'ns2.example.com.',
            'ns3.example.com.',
            'ns4.example.com.'
        ]
});

// Domains:
D('example.com', REG_NONE, DnsProvider(DNS_BIND),
    DefaultTTL('30m'),
    NAMESERVER_TTL('1h'),
    A('@', '10.10.10.10'),
    A('localhost', '127.0.0.1'),
    NAPTR("foo.com",100,10,"U","E2U+sip","!^.*$!sip:customer-service@example.com!","localhost")
);

$ dnscontrol preview  
Executing javascript in dnsconfig.js: ReferenceError: 'NAPTR' is not defined
```

Can you please help me figuring out what's missing?

### CODE REVIEW

**Comment by tlimoncelli** [comment]
Could you check to see if you followed Step 3 in  https://stackexchange.github.io/dnscontrol/adding-new-rtypes ?  (especially the FYI)

HTH

**Comment by karlism** [comment]
@tlimoncelli, yes, it seems to be there:
https://github.com/StackExchange/dnscontrol/pull/461/files#diff-a9e9ad8f8187b1ff7eb56ad0b3206529R220
I also regenerated and updated `static.go`:
https://github.com/StackExchange/dnscontrol/pull/461/files#diff-c0c7c2de9997c4dc5ecbd86fe0fd3641

**Comment by tlimoncelli** [comment]
Try this:

> dnscontrol preview --dev

**Comment by karlism** [comment]
@tlimoncelli, thanks, I've found the issue. Now records are being created properly, but there is another issue - it detects changes every time when I push it despite the fact that dnsconfig.js is intact:
```
$ cat dnsconfig.js                                                                                                                                                                                                                   
// Providers:
var REG_NONE = NewRegistrar('none', 'NONE');    // No registrar.
var DNS_BIND = NewDnsProvider('bind', 'BIND', {
        'default_soa': {
            'master': 'master.example.com.',
            'mbox': 'hostmaster.example.com.',
            'refresh': 3600,
            'retry': 600,
            'expire': 604800,
            'minttl': 1440,
        },
        'default_ns': [
            'ns1.example.com.',
            'ns2.example.com.',
            'ns3.example.com.',
            'ns4.example.com.'
        ]
});

// Domains:
D('example.com', REG_NONE, DnsProvider(DNS_BIND),
    DefaultTTL('30m'),
    NAMESERVER_TTL('1h'),
    A('test', '127.0.0.1'),
    NAPTR('internet.1stmobile.apn.epc', 1, 100, 'A', 'x-3gpp-pgw:x-s5-gtp:x-s8-gtp:x-gn:x-gp', '', 'test')
);
petruha$ rm -rf zones/example.com.zone  
petruha$ dnscontrol push                
INFO: Config file "creds.json" does not exist. Skipping.
******************** Domain: example.com
----- Getting nameservers from: bind
----- DNS Provider: bind...1 correction
#1: GENERATE_ZONEFILE: example.com
 (7 records)

CREATING ZONEFILE: zones/example.com.zone
SUCCESS!
----- Registrar: none...0 corrections
Done. 1 corrections.
petruha$ dnscontrol push                
INFO: Config file "creds.json" does not exist. Skipping.
******************** Domain: example.com
----- Getting nameservers from: bind
----- DNS Provider: bind...1 correction
#1: GENERATE_ZONEFILE: example.com
MODIFY NAPTR internet.1stmobile.apn.epc.example.com: (1 100 "A" "x-3gpp-pgw:x-s5-gtp:x-s8-gtp:x-gn:x-gp" "" test.example.com. ttl=1800) -> (1 100 "A" "x-3gpp-pgw:x-s5-gtp:x-s8-gtp:x-gn:x-gp" "" test ttl=1800)

CREATING ZONEFILE: zones/example.com.zone
SUCCESS!
----- Registrar: none...0 corrections
Done. 1 corrections.
```
Any ideas why `dnscontrol` thinks the record has changed?

**Comment by tlimoncelli** [comment]
It looks like when the old and new are compared, one has 'test' and the other has 'test.example.com.". 

I recommend both should have the FQDN.  IIRC we tried both with MX and keeping everything FQDN was the best.

**Comment by karlism** [comment]
@tlimoncelli, thanks, it seems to be working correctly now. Can you please review this merge request and merge it if everything is OK?

File: models/record.go [hunk 1]
```diff
@@ -75,6 +76,13 @@ type RecordConfig struct {
 	SrvPort          uint16            `json:"srvport,omitempty"`
 	CaaTag           string            `json:"caatag,omitempty"`
 	CaaFlag          uint8             `json:"caaflag,omitempty"`
+	NaptrOrder       uint16            `json:"naptrorder,omitempty"`
+	NaptrPreference  uint16            `json:"naptrpreference,omitempty"`
+	NaptrFlags       string            `json:"naptrflags,omitempty"`
+	NaptrService     string            `json:"naptrservice,omitempty"`
+	NaptrRegexp      string            `json:"naptrregexp,omitempty"`
+	//NaptrReplacement string            `json:"naptrreplacement,omitempty"`
```
Comment (tlimoncelli) [inline models/record.go]: Please include a comment explaining why these are commented out.

**Comment by tlimoncelli** [comment]
When I run the integration tests I get a failure:

```
cd ~/src/github.com/StackExchange/dnscontrol/integrationTest && go test -v -verbose -provider BIND
...
...
...
...
        --- PASS: TestDNSProviders/example.com/39:_Empty (0.00s)
            integration_test.go:136: GENERATE_ZONEFILE: example.com
                DELETE PTR 4.example.com bar.com. ttl=300
                
        --- FAIL: TestDNSProviders/example.com/40:_NAPTR_record (0.00s)
            integration_test.go:136: GENERATE_ZONEFILE: example.com
                CREATE NAPTR example.example.com 100 10 "U" "E2U+sip" "!^.*$!sip:customer-service@example.com!" foo.com ttl=300
                
            integration_test.go:149: Expected 0 corrections on second run, but found 1.
            integration_test.go:151: #0: GENERATE_ZONEFILE: example.com
                MODIFY NAPTR example.example.com: (100 10 "U" "E2U+sip" "!^.*$!sip:customer-service@example.com!" foo.com.example.com. ttl=300) -> (100 10 "U" "E2U+sip" "!^.*$!sip:customer-service@example.com!" foo.com ttl=300)
```

**Comment by tlimoncelli** [comment]
Can you reproduce the error I got?

File: models/record.go (hunk 1 above)
Comment (karlism) [reply models/record.go]: These comments have been removed together with `models/.record.go.swp` swapfile commited by accident.

**Comment by karlism** [comment]
@tlimoncelli, yes I can. I've been busy lately, I'll try to fix integration tests next week.

**Comment by karlism** [comment]
@tlimoncelli, can you please check again?

**Comment by tlimoncelli** [comment]
Looks great!  Thank you for adding this much-needed feature!

**Pull request merged**